common/display/displayqueue.cpp

```diff
@@ -686,6 +686,11 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
     }
   }
 
+  if (previous_size != size) {
```

Comment:
Yes, it may introduce minor performance impact for frame that visible layers changed from last frame. 

For black stripe issue, as introduced in previous comment, it may introduced by improper object initialization & linkage.